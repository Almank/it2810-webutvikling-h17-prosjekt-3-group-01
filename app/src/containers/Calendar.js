import React from 'react';
import { View, StyleSheet, AsyncStorage, Text, ScrollView } from 'react-native';
import {Week} from '../components/Calendar/Week';
import {AppointmentForm} from "../components/Calendar/AppointmentForm";

export class Calendar extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            dateToday: new Date().toISOString().slice(0, 10),
            dateVal: '',
            timeVal: '',
            titleVal: '',
            textVal: '',
            children: [
                {date: '2017-10-15', time: '14:15', title: 'Hello World', text: 'This is text'},
            ],
        };

        //Binding functions
        this.createAppointment = this.createAppointment.bind(this);
        this.changeContent = this.changeContent.bind(this);

        this.loadData();
    }

    emptyScheduleCheck(){

        //Copy of the state children array.
        let myData =[].concat(this.state.children)

        //Sorting elements based on time, earliest first.
            .sort((a, b) => a.time > b.time)

            //Filtering out appointments on current day.
            .filter(child => child.date === this.state.dateToday)

            //Mapping items from array giving the html the correct values.
            .map((item,i) =>
                <View style={styles.showAppointments} key={i}>
                    <Text style={styles.appointmentItem}>{item.time}</Text>
                    <Text style={styles.appointmentItem}>{item.title}</Text>
                    <Text style={styles.appointmentItem}>{item.text}</Text>
                </View>
            );

        //If there are no plans that day.
        if (myData.length === 0){
            myData = [  <View style={styles.showAppointments} key={0}>
                <Text style={styles.appointmentNoItems}>You don't have any plans!</Text>
            </View>];
        }

        return (
            <ScrollView style={styles.scrollView}>
                {myData}
            </ScrollView>
        );
    }

    changeContent(e){
        let data = this.state;
        data.dateToday = e.props.dateFull;
        this.setState({ dateToday: e.props.dateFull }, () => {
            this.forceUpdate();
        });
        try{
            AsyncStorage.setItem("calendar", JSON.stringify(data));
        } catch (error){
            console.log(error);
        }
    }

    async loadData(){
        let data;
        try {
            data = await AsyncStorage.getItem('calendar');
            if (data !== null){
                data = JSON.parse(data);
                data.dateToday = new Date().toISOString().slice(0, 10);
                this.setState(data);
            } else{
                this.setState({
                    dateToday: new Date().toISOString().slice(0, 10),
                    children: [],
                    dateValue: '',
                    timeValue: '',
                    titleValue: '',
                    textValue: '',
                });
            }
        } catch (error) {
            console.log(error);
        }
    }

    createAppointment(e) {
        //Fetching values from form.
        let dateValue = e[0];
        let timeValue = e[1];
        let titleValue = e[2];
        let textValue = e[3];

        //Validate fields date, time and title. Last field is optional.
        if (this.validateFormDate(dateValue) && this.validateFormTime(timeValue) && this.validateFormTitle(titleValue)){

            //Creating new array with current state children.
            let newStateArray = this.state.children.slice();

            //Pushing new child to array.
            newStateArray.push({date: dateValue, time: timeValue, title: titleValue, text: textValue});

            //Sorting array with earliest appointments first.
            newStateArray.sort((a, b) => a.time > b.time);

            //Creating new updated state.
            let data = {
                children: newStateArray,
                dateToday: new Date().toISOString().slice(0, 10),
            };

            //Setting state.
            this.setState(data);

            //Setting updated values to storage.
            try{
                AsyncStorage.setItem("calendar", JSON.stringify(data));
            } catch (error){
                console.log(error);
            }

        } else {
            //If there are invalid values.
            alert('Error! Do not submit invalid values.')
        }
    }

    validateFormDate(date){
        return (date.length === 10)
    }

    validateFormTime(time){
        return (time.length === 5)
    }

    validateFormTitle(title){
        return (title.length > 0)
    }

    render(){
        return(
            <View style={styles.container}>
                <Week change={this.changeContent} />
                <View style={styles.bottomContainer}>
                    <AppointmentForm getValues={ arr => this.createAppointment(arr) } />
                    {this.emptyScheduleCheck()}
                </View>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        marginTop: 40,
        display: 'flex',
        flexDirection: 'column',
        flex: 1,
        marginBottom: 20,
    },
    bottomContainer: {
        display: 'flex',
        flexDirection: 'column',
        flex: 1,
    },
    showAppointments: {
        display: 'flex',
        flexDirection: 'row',
        justifyContent: 'center',
    },
    appointmentItem: {
        flexBasis:'33%',
        textAlign:'center',
        fontSize:16,
        flexWrap: 'wrap',
    },
    appointmentNoItems: {
        marginTop: '10%',
        flexBasis:'90%',
        textAlign:'center',
        fontSize:20,
    },
    scrollView: {
        height: '100%',
    }


});



