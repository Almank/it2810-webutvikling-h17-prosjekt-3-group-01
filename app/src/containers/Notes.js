import React from 'react';
import { View, Text } from 'react-native';

export class Notes extends React.Component {
    render(){
        return(
            <Text>Notes</Text>
        )
    }
}