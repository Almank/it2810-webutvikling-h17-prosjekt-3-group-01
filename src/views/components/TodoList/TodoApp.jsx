import React from 'react';
import PropTypes from 'prop-types';
import ReactDOM from 'react-dom';
import ToDoBanner from './TodoBanner';
import ToDoForm from './TodoForm';
import ToDoList from './TodoList';
import ToDoListItem from './TodoListItem';
import ToDoFilter from './TodoFilter';
import ToDoCatelog from './TodoCatelog';
import ToDoCatelogForm from './TodoCatelogForm';

/* [TODO APP] */ 
var ToDoApp = React.createClass({
		getInitialState : function(){
			if(localStorage.getItem("todo")=== null){
                let data = {Todo:[{name:"Primary",items:[{item:'Todo item #1',isDone:false},{item:'Todo item #2',isDone:true},{item:'aaaa',isDone:true},{item:'dddd',isDone:true}
			]},{name:"Secondary",items:[{item:'Todo item #1',isDone:false},{item:'Todo item #2',isDone:true},{item:'Todo item #3',isDone:true}
			]}],filter:[{keyword:'',Status:"SHOW_ALL"}],selectedItem:"0"};
                localStorage.setItem("todo", JSON.stringify(data));
            }
            let todo = localStorage.getItem("todo");
            todo = JSON.parse(todo);
            console.log(todo);
            return this.state = {
                Todo: todo['Todo']
            }
            
		},
		updateItems: function(newItem){	
			var item = {item:newItem,isDone:false};
			var newtodo = this.state.Todo;
			var allItems = this.state.Todo[this.state.selectedItem].items.concat([item]);
			newtodo[this.state.selectedItem].items = allItems;
			this.setState({
				Todo: newtodo
			});
            localStorage.setItem("todo", JSON.stringify(this.state.Todo));
		},
		deleteItem : function(index){
			var newtodo = this.state.Todo;
			var allItems = this.state.Todo[this.state.selectedItem].items.slice(); //copy array
			allItems.splice(index, 1); //remove element
			newtodo[this.state.selectedItem].items = allItems;
			this.setState({
				Todo: newtodo
			});
		},
		filterItem : function(e){
			
			this.state.filter[0].Status = e.target.value;
			this.setState({
				filter: this.state.filter
			});
		},
		searchItem : function(e){
	
			this.state.filter[0].keyword = e.target.value;
			this.setState({
				filter: this.state.filter
			});
		},
		AddCatalog: function(newCatalog){
			var Catalog = {name:newCatalog,items:[{item:'Todo item #1',isDone:false}]};
			var newtodo = this.state.Todo.concat([Catalog]);
			this.setState({
				Todo: newtodo
			});
		},
		setSelectedCatalog: function(index){
			this.state.selectedItem = index;
			this.setState({
				selectedItem: index
			});
		},
		render: function(){
			return (
				<div className="row">
					<div className="col-xs-3">
                        <ToDoCatelogForm onFormSubmit = {this.AddCatalog} />
                        <ToDoCatelog selectedID = {this.state.selectedItem} onSelected={this.setSelectedCatalog} Todos = {this.state.Todo} />
					</div>
					<div className="col-xs-6">
						<ToDoBanner/>
						<ToDoFilter onFilter = {this.filterItem} onSearch = {this.searchItem} filter={this.state.filter}/>
						<ToDoForm onFormSubmit = {this.updateItems} />
						<ToDoList  items = {this.state.Todo[this.state.selectedItem].items} filter = {this.state.filter} onDelete={this.deleteItem}/>
					</div>
				</div>
			);
		}
	});

export default ToDoApp;