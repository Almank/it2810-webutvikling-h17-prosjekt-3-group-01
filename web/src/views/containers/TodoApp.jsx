import React from 'react';
import '../../assets/styles/Todolists.css';
import {TodoForm, TodoList, TodoFilter, TodoCatelog, TodoCatelogForm} from '../Components';

/* [TODO APP] */
export class ToDoApp extends React.Component{
	constructor(props){
		super(props);

		if(localStorage.getItem("todo") === null){
			let data = {Todo: [],
								selectedItem: "0"};
			localStorage.setItem("todo", JSON.stringify(data));
		}
		let data = {Todo: [{name:"Primary",
			items:[]},],
			filter: [{keyword:'',Status:"SHOW_ALL"}],
			selectedItem: "0"};
		localStorage.setItem("todo", JSON.stringify(data));
		let todo = localStorage.getItem("todo");
		todo = JSON.parse(todo);
		this.state = todo;

		this.updateItems = this.updateItems.bind(this);
		this.deleteItem = this.deleteItem.bind(this);
		this.filterItem = this.filterItem.bind(this);
		this.AddCatalog = this.AddCatalog.bind(this);
		this.searchItem = this.searchItem.bind(this);
		this.filterItem = this.filterItem.bind(this);
		this.setSelectedCatalog = this.setSelectedCatalog.bind(this);
	}

	updateItems(newItem){
		let item = {item:newItem,isDone:false};
		let newtodo = this.state;
		let allItems = this.state.Todo[this.state.selectedItem].items.concat([item]);
		newtodo.Todo[this.state.selectedItem].items = allItems;
		this.setState(newtodo);
		localStorage.setItem("todo", JSON.stringify(this.state));
	}

	deleteItem(index){
		let newtodo = this.state.Todo;
		let allItems = this.state.Todo[this.state.selectedItem].items.slice(); //copy array
		allItems.splice(index, 1); //remove element
		newtodo[this.state.selectedItem].items = allItems;
		this.setState(newtodo);
		localStorage.setItem("todo", JSON.stringify(this.state));
	}

	filterItem(e){
		let filter = this.state.filter;
		filter[0].Status = e.target.id;
		this.setState({
			filter: filter,
		});
	}

	searchItem(e){
		let filter = this.state.filter;
		filter[0].keyword = e.target.value;
		this.setState({
			filter: filter,
		});
	}

	AddCatalog(newCatalog){
		let Catalog = {name:newCatalog,items:[{item:'Todo item #1',isDone:false}]};
		let newtodo = this.state.Todo.concat([Catalog]);
		this.setState({Todo: newtodo});
		localStorage.setItem("todo", JSON.stringify(this.state));
	}

	setSelectedCatalog(index){
		this.setState({
			selectedItem: index,
		});
	}

	render(){
		return (
			<div className="todo-wrapper">
				<h1 className={'title'} >To-do list</h1>
				<div className="bigrow">
					<div className="col1">
						<h3>To-do lists:</h3>
						<TodoCatelogForm onFormSubmit = {this.AddCatalog} />
						<TodoCatelog selectedID={this.state.selectedItem} onClick={this.setSelectedCatalog} Todos = {this.state.Todo} />
					</div>
					<div className="col2">
						<h3>What do you want to-do?</h3>
						<TodoFilter onFilter = {this.filterItem} onSearch = {this.searchItem} filter={this.state.filter}/>
						<TodoForm onFormSubmit = {this.updateItems} />
						<TodoList  items = {this.state.Todo[this.state.selectedItem].items} filter = {this.state.filter} onDelete={this.deleteItem}/>
					</div>
				</div>
			</div>
		);
	}

}