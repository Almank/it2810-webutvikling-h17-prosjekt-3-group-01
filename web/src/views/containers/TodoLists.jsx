/**
 * Created by martinlunde on 27.09.2017.
 */
import React from 'react';
import ToDoApp from "../components/TodoList/TodoApp";


export class TodoLists extends React.Component {
    render(){
        return ( 
            <div id="todolist">
                <ToDoApp/>
            </div>

        );


    }
}