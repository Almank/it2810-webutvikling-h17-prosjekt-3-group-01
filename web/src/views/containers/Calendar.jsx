/**
 * Created by almank on 02.10.2017.
 */
import React from 'react';
import {WeekCal, AppointmentForm, ContentHeader} from '../Components';
import '../../assets/styles/Calendar.css';

export class Calendar extends React.Component {
    constructor(props) {

        //Setting state if there is none stored.
        if(localStorage.getItem("emptyCalendar") === null){
            let data = {
                dateToday: new Date().toISOString().slice(0, 10),
                children: [],
            };
            localStorage.setItem("emptyCalendar", JSON.stringify(data));
        }
        super(props);

        //Fetching state from storage.
        let data = localStorage.getItem("emptyCalendar");
        data = JSON.parse(data);

        //Setting dateToday to current date.
        data.dateToday = new Date().toISOString().slice(0, 10);

        //Setting state to stored state.
        this.state = data;

        //Binding functions
        this.createAppointment = this.createAppointment.bind(this);
        this.changeContent = this.changeContent.bind(this);
        this.handleRemoveClick = this.handleRemoveClick.bind(this);
        this.updateLocalStorage = this.handleRemoveClick.bind(this);
        this.handleChangeClick = this.handleChangeClick.bind(this);
    }

    handleRemoveClick(event){
        for (let i=0; i<this.state.children.length; i++) {
            if (String(this.state.children[i].uniqueDate) === String(event.target.value)){
                //Deleting values in child
                delete this.state.children[i];

                //Removing empty objects
                let newArr = this.state.children.filter(val => Object.keys(val).length !== 0);

                //Updating state and localstorage on with callback function.
                this.setState({children: newArr}, function() {
                    localStorage.setItem("emptyCalendar", JSON.stringify(this.state));
                });
            }
        }
    }

    handleChangeClick(type, event){
        for (let i=0; i<this.state.children.length; i++) {
            if (String(this.state.children[i].uniqueDate) === String(event.target.name)) {
                let data = this.state.children;
                if (type === 'title')
                    data[i].title = event.target.value;
                else if (type === 'time')
                    data[i].time = event.target.value;
                else if (type === 'text')
                    data[i].text = event.target.value;
                this.setState({children: data});
                localStorage.setItem("emptyCalendar", JSON.stringify(this.state));
            }
        }
    }

    emptyScheduleCheck(){
        //Copy of the state children array.
        let myData =[].concat(this.state.children)

            //Filtering out appointments on current day.
            .filter(child => child.date === this.state.dateToday)

            //Sorting elements based on time, earliest first.
            .sort((a, b) => parseInt((("" + a.time.slice(0,2)) + a.time.slice(3,6)), 0) -
                            parseInt(("" + b.time.slice(0,2)) + b.time.slice(3,6), 0))

            //Mapping items from array giving the html the correct values.
            .map((item,i) =>
            <div className='calendarBoxContent' key={i}>
                <div className='leftCalendarBoxContent'>
                    <input onBlur={this.handleChangeClick.bind(this, 'time')}
                           type='time'
                           defaultValue={this.state.children.filter(a => a.uniqueDate === item.uniqueDate)[0].time}
                           name={item.uniqueDate}
                           className='changeField'/>
                </div>

                <div className='middleCalendarBoxContent'>
                    <input onBlur={this.handleChangeClick.bind(this, 'title')}
                           type='text'
                           defaultValue={this.state.children.filter(a => a.uniqueDate === item.uniqueDate)[0].title}
                           name={item.uniqueDate}
                           maxLength={10}
                           className='changeField'/>
                </div>
                <div className='rightCalendarBoxContent'>
                    <textarea onBlur={this.handleChangeClick.bind(this, 'text')}
                           type='text'
                           defaultValue={this.state.children.filter(a => a.uniqueDate === item.uniqueDate)[0].text}
                           name={item.uniqueDate}
                           maxLength={200}
                           className='changeField'/>
                </div>
                <button onClick={this.handleRemoveClick}
                        className='RemoveButton'
                        type='button'
                        value={item.uniqueDate}>
                    <span className='glyphicon glyphicon-minus' />
                </button>
            </div>
            );
        //If there are no plans that day.
        if (myData.length === 0){
            myData = [  <div key={0} className='calendarBoxContent'>
                            <h1>You don't have any plans!</h1>
                        </div>];
            }

        return (
            <div className='appointmentFlow'>
                {myData}
            </div>
        );
    }

    createAppointment(e) {
        //Preventing reload of window.
        e.preventDefault();

        //Fetching values from form.
        let dateValue = document.getElementsByClassName('dateInput')[0].value;
        let timeValue = document.getElementsByClassName('timeInput')[0].value;
        let titleValue = document.getElementsByClassName('titleInput')[0].value;
        let textValue = document.getElementsByClassName('textInput')[0].value;

        //Validate fields date, time and title. Last field is optional.
        if (this.validateFormDate(dateValue) && this.validateFormTime(timeValue) && this.validateFormTitle(titleValue)){

            //Creating new array with current state children.
            let newStateArray = this.state.children.slice();

            //Pushing new child to array.
            newStateArray.push({date: dateValue, time: timeValue, title: titleValue, text: textValue, uniqueDate: new Date()});

            //Sorting array with earliest appointments first.
            newStateArray.sort((a, b) =>    parseInt(("" + a.time.slice(0,2)) + a.time.slice(3,6), 0) -
                                            parseInt(("" + b.time.slice(0,2)) + b.time.slice(3,6), 0));

            //Creating new updated state.
            let data = {
                children: newStateArray,
                dateToday: new Date().toISOString().slice(0, 10),
            };

            //Setting state.
            this.setState(data);

            //Updating localstorage to store new data.
            localStorage.setItem("emptyCalendar", JSON.stringify(data));

            //Resetting form values.
            document.getElementsByClassName('dateInput')[0].value = null;
            document.getElementsByClassName('timeInput')[0].value = null;
            document.getElementsByClassName('titleInput')[0].value = null;
            document.getElementsByClassName('textInput')[0].value = null;

            //Hiding form.
            this.showForm();
        } else {
            alert('Invalid values. Please try again.')
        }
    }

    validateFormDate(date){
        return date.length === 10 && date >= new Date().toISOString().slice(0, 10);
    }

    validateFormTime(time){
        return time.length === 5;
    }

    validateFormTitle(title){
        return title.length > 0;
    }

    changeContent(e){
        //Fetching data from localstorage
        let data = localStorage.getItem("emptyCalendar");
        data = JSON.parse(data);

        //Setting dateToday to current date.
        data.dateToday = e.props.dateFull;

        //Setting state to stored state.
        this.state = data;

        //Updating localstorage to store new data.
        localStorage.setItem("emptyCalendar", JSON.stringify(data));

        //Forcing update of the DOM to change content.
        this.forceUpdate();
    }

    showForm(){
        //Toggle form.
        let element = document.querySelector('.formContainer');
        element.style.display = element.style.display === 'none' ? 'flex' : 'none';
        let today = new Date().toISOString().slice(0, 10);
        document.querySelector(".dateInput").value = today;
        let fields = document.querySelectorAll(".inputField");
        for (let i = 0; i < fields.length; i++)
            fields[i].style.border = '1px solid lightgray';

    }

    render(){
        return (
            <div className={'calendarContent'}>
                <h1 className={'title'} >Your schedule for the coming week</h1>
                <WeekCal change={this.changeContent} />

                <div className={'calendarBox'}>
                    <ContentHeader closeForm={this.showForm}/>
                    <AppointmentForm closeForm={this.showForm} submitForm={this.createAppointment}/>
                    {this.emptyScheduleCheck()}
                </div>
            </div>
        );
    }
}