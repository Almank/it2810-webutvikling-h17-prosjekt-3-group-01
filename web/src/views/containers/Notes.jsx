/**
 * Created by martinlunde on 27.09.2017.
 */
import React from 'react';
import '../../assets/styles/Notes.css';
import {NoteList, NoteContent} from "../Components";
import {Route} from 'react-router-dom';


export class Notes extends React.Component {
    constructor(props){
        //Initiate local storage if no data is stored.
        if(localStorage.getItem("notes") === null){
            let data = {
                notes: {
                    'default': {
                        title: 'default',
                        content: 'no content'
                    }
                },
                update: false
            };
            localStorage.setItem("notes", JSON.stringify(data));
        }

        //Load locally stored data into state.
        let notes = localStorage.getItem("notes");
        notes = JSON.parse(notes);
        super(props);
        this.state = {
            notes: notes['notes'],
        };
        this.removeClick = this.removeClick.bind(this);
    };

    //Function for updating locally stored data.
    updateLocalStorage(){
        let data = this.state;
        localStorage.setItem("notes", JSON.stringify(data));
    }

    //Function for updating a notes title and content.
    onChange(title, content){
        // eslint-disable-next-line
        this.state.notes[title].title = title;
        // eslint-disable-next-line
        this.state.notes[title].content = content;
        this.updateLocalStorage();
    }

    //Function for enabling remove button and updating storage.
    removeClick(title){
        delete this.state.notes[title];
        this.updateLocalStorage();
        this.setState({update: true});
        window.location.pathname = "/notebook";
    }

    render(){
        return (
            <div className="NoteBookWrapper">
                <h1 className={'title'}> Notes</h1>
                <div className="NoteBook">
                    <NoteList notes={this.state.notes}
                              removeClick={(value) => this.removeClick(value)}
                              updateStorage={() => this.updateLocalStorage()}
                    />
                    <div className="NoteEdit">
                        <div className="NoteEditContent">
                            <Route exact path="/notebook/:name" render={ (id) =>
                                <NoteContent data={this.state.notes[id.match.params.name]}
                                             key={id}
                                             id={id}
                                             onChange={this.onChange.bind(this)}
                                             update={this.state.update}
                                />}/>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}