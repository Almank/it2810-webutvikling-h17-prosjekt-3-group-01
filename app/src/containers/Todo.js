import React from 'react';
import {TodoList} from "../components/TodoList/TodoList";

export class Todo extends React.Component {
    render(){
        return(
            <TodoList />
        )
    }
}