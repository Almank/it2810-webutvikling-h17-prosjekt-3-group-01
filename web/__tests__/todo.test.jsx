import React from 'react';
import localStorag from '../__test_setup__/localstorage';

import {ToDoApp} from "../src/views/containers/TodoApp";

const RENDER = render(<ToDoApp/>);
const INST_TO_DO_APP = shallow(<ToDoApp/>).instance();


describe("TodoApp", () => {
    it("Should render correctly", () => {
        expect(RENDER).toMatchSnapshot();
    });

    describe("TodoApp.addCatalog()", () => {
        it("Should add a new catalog to state", () => {
            const PREV_LENGTH = INST_TO_DO_APP.state.Todo.length;
            INST_TO_DO_APP.AddCatalog("test");
            expect(INST_TO_DO_APP.state.Todo.length).toBe(PREV_LENGTH + 1);
        })
    });
    describe("TodoApp.setSelectedCatalog()", () => {
        it("Should set the selected catalog correctly based on the input index", () => {
            INST_TO_DO_APP.setSelectedCatalog(0);
            expect(INST_TO_DO_APP.state.selectedItem).toBe(0);
        })
    });
    describe("TodoApp.updateItems()", () => {
        it("Should add a new item to correct catalog", () => {
            const PREV_LENGTH = INST_TO_DO_APP.state.Todo[INST_TO_DO_APP.state.selectedItem].items.length;
            INST_TO_DO_APP.updateItems("newTestItem");
            expect(INST_TO_DO_APP.state.Todo[INST_TO_DO_APP.state.selectedItem].items.length).toBe(PREV_LENGTH + 1);
        });
    });
    describe("TodoApp.deleteItem()", () => {
        it("Should delete item from correct catalog", () => {
            INST_TO_DO_APP.updateItems("itemToBeDeleted");
            const PREV_LENGTH = INST_TO_DO_APP.state.Todo[INST_TO_DO_APP.state.selectedItem].items.length;
            INST_TO_DO_APP.deleteItem(PREV_LENGTH - 1);
            expect(INST_TO_DO_APP.state.Todo[INST_TO_DO_APP.state.selectedItem].items.length).toBe(PREV_LENGTH - 1);

        })
    })


});