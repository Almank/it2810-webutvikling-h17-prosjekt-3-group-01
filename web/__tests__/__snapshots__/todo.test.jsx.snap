// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoApp Should render correctly 1`] = `
<div
  class="todo-wrapper"
>
  <h1
    class="title"
  >
    To-do list
  </h1>
  <div
    class="bigrow"
  >
    <div
      class="col1"
    >
      <h3>
        To-do lists:
      </h3>
      <div
        class="row"
      >
        <form>
          <div
            class="form-group "
          >
            <input
              class="todoField"
              placeholder="Add list"
              type="text"
              value=""
            />
            <input
              class="addbutton"
              type="submit"
              value="Add"
            />
          </div>
        </form>
      </div>
      <div
        class="list-group"
      >
        <a
          class="list-group-item "
          data-id="0"
          href="#"
        >
          <span
            class="badge"
          >
            4
          </span>
          Primary
        </a>
        <a
          class="list-group-item "
          data-id="1"
          href="#"
        >
          <span
            class="badge"
          >
            3
          </span>
          Secondary
        </a>
      </div>
    </div>
    <div
      class="col2"
    >
      <h3>
        What do you want to-do?
      </h3>
      <div
        class="row"
      >
        <form>
          <div
            class="todoform"
          >
            <input
              class="todoField"
              placeholder="Add to-do"
              type="text"
              value=""
            />
            <input
              class="addbutton"
              type="submit"
              value="Add"
            />
          </div>
        </form>
      </div>
      <ul>
        <li
          data-id="0"
          id="todoitem"
        >
          <button
            aria-hidden="true"
            class="close pull-right"
            type="button"
          >
            x
          </button>
          <input
            id="check"
            type="checkbox"
          />
          <span
            style="text-decoration:none"
          >
            Todo item #1
          </span>
        </li>
        <li
          data-id="1"
          id="todoitem"
        >
          <button
            aria-hidden="true"
            class="close pull-right"
            type="button"
          >
            x
          </button>
          <input
            checked=""
            id="check"
            type="checkbox"
          />
          <span
            style="text-decoration:line-through"
          >
            Todo item #2
          </span>
        </li>
        <li
          data-id="2"
          id="todoitem"
        >
          <button
            aria-hidden="true"
            class="close pull-right"
            type="button"
          >
            x
          </button>
          <input
            checked=""
            id="check"
            type="checkbox"
          />
          <span
            style="text-decoration:line-through"
          >
            aaaa
          </span>
        </li>
        <li
          data-id="3"
          id="todoitem"
        >
          <button
            aria-hidden="true"
            class="close pull-right"
            type="button"
          >
            x
          </button>
          <input
            checked=""
            id="check"
            type="checkbox"
          />
          <span
            style="text-decoration:line-through"
          >
            dddd
          </span>
        </li>
      </ul>
      <div
        class="row"
      >
        <div
          class="col-filter"
        >
          <ul
            class="todo-filter"
          >
            <li>
              <a
                class="button button"
                href="#"
                id="SHOW_ALL"
              >
                All
              </a>
            </li>
            <li>
              <a
                class="button "
                id="false"
              >
                Incomplete
              </a>
            </li>
            <li>
              <a
                class="button "
                id="true"
              >
                Complete
              </a>
            </li>
          </ul>
        </div>
        <div
          class="col-search"
        >
          <div
            class="input-group"
          >
            <span
              class="input-group-btn"
            >
              <button
                class="button"
                type="button"
              >
                <span
                  class="glyphicon glyphicon-search"
                />
              </button>
            </span>
            <input
              class="todoField"
              placeholder="Search"
              type="search"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
